lib/Styler/widgets/FilterBuilder.js

```diff
@@ -1155,6 +1201,81 @@ Styler.FilterBuilder = Ext.extend(Ext.Panel, {
+                "||": Styler.FilterBuilder.ANY_OF,
+                "!": Styler.FilterBuilder.NONE_OF
+            };
+            //filter.type mapped to Styler.FilterBuilder combobox value
+            var comboType = {
+                "&&": 1,
+                "||": 0,
+                "!": 2
+            };
+
+            this.addCondition("group");
+
+            //we remove empty gx_filterpanel from last gx_filterbuilder
+            var gfb = this.findByType("gx_filterbuilder")[this.findByType("gx_filterbuilder").length - 1];
+            //from removeCondition method
+            var parent = gfb.filter.filters[0].filters;
+            parent.splice(0, 1);
+            //remove associate gx_filterpanel (fisrt component of the childFiltersPanel)
+            gfb.childFiltersPanel.remove(gfb.childFiltersPanel.getComponent(0))
+
+            if (!(stylerType[filter.type] === undefined)) {
+                this._dbg("generic case, Combo and Builder Type set to" + filter.type);
+
+                this.findByType("gx_filterbuilder")[this.findByType("gx_filterbuilder").length -
+                1].findByType("combo")[0].setValue(comboType[filter.type]);
+                this.findByType("gx_filterbuilder")[this.findByType("gx_filterbuilder").length -
+                1].changeBuilderType(stylerType[filter.type]);
+            }
+
+            var gxFilterBuilderChildren = this.childFiltersPanel.findByType("gx_filterbuilder");
+            Ext.each(filter.filters, function(filter) {
+                this._dbg("generic case, call _loadfilter on filter of type " + filter.type);
+                this._loadfilters(filter);
+            }, gxFilterBuilderChildren[gxFilterBuilderChildren.length -
+            1]);
+        }
+        if (filter instanceof OpenLayers.Filter.Comparison) {
+            this._dbg("generic case, adding a comparison condition");
+
+            this.addCondition("default", null, filter);
+        }
+        if (filter instanceof OpenLayers.Filter.Spatial) {
+            if (!this.allowSpatial) {
+                alert("Adding spatial filter to non-spatial filter builder");
+                return;
+            }
+
+            this._dbg("generic case, adding a spatial condition");
+
+            var feature = new OpenLayers.Feature.Vector(
+                filter.value, {}, this.vectorLayer.style);
+            // Adding a feature to vectorLayer fire an addCondition
+            // without the filter parameter
+            // We manually remove the filter
+            // and the row
+            this.vectorLayer.addFeatures([feature]);
+            this.childFiltersPanel.remove(this.childFiltersPanel.items.items[this.childFiltersPanel.items.getCount() -
+            1].getId());
+            this.filter.filters[0].filters.pop();
+            this.addCondition("spatial", feature, filter);
+        }
+    },
+
+    _dbg: function(msg) {
+        if (this._debug) {
+            console.log('DEBUG - ' + msg);
```

Comment:
IIRC OpenLayers.Console (https://github.com/georchestra/openlayers/blob/master/lib/OpenLayers/Console.js) was made for this case, but I'm not sure this is worth changing your code...